internal/api.go

```diff
@@ -114,8 +119,17 @@ func (a *defaultAPI) ToGraphiteName(metric api.TaggedMetric) (api.GraphiteMetric
 }
 
 func (a *defaultAPI) ToTaggedName(metric api.GraphiteMetric) (api.TaggedMetric, error) {
+	a.mutex.Lock()
+	if result, ok := a.cached[metric]; ok {
+		a.mutex.Unlock()
+		return result, nil
+	}
+	a.mutex.Unlock()
 	match, matched := a.ruleset.MatchRule(string(metric))
 	if matched {
+		a.mutex.Lock()
+		a.cached[metric] = match
```

Comment:
failure should be cached also - otherwise failed run will continuously burn CPU.